llvm/lib/SYCLLowerIR/ESIMD/ESIMDVerifier.cpp

```diff
@@ -136,7 +137,13 @@ class ESIMDVerifierImpl {
             return LegalNameRE.match(Name);
           };
           if (any_of(LegalSYCLFunctions, checkLegalFunc) ||
-              (ForceStatelessMem &&
+              // TODO: Methods listed in LegalSYCLFunctionsInStatelessMode are
+              // indeed required to support ESIMD APIs accepting accessors in
+              // stateless-only mode. This unintentionally opens that API for
+              // unintended usage in user's programs. Can those APIs be
```

Comment:
Not sure if I understand the issue but we could write a pass (or even do it here) to error if they're used in functions not marked esimd. Obviously this would be seperate work outside of this PR

Reply:
If you don't mind, I would do that in a separate PR. For now I plan to create an internal ticket for this issue/TODO.

Reply:
Sure, sounds good